org.osgi.test.junit5.cm/src/test/java/org/osgi/test/junit5/cm/ConfigAnnotationTest.java

```diff
@@ -0,0 +1,182 @@
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.osgi.test.common.annotation.config.ConfigEntry;
+import org.osgi.test.common.annotation.config.InjectConfiguration;
+import org.osgi.test.common.annotation.config.WithConfiguration;
+import org.osgi.test.common.annotation.config.WithFactoryConfiguration;
+
+@ExtendWith(ConfigurationExtension.class)
+@WithConfiguration(pid = ConfigAnnotationTest.NONSTATIC_CONFIGURATION_PID)
+public class ConfigAnnotationTest {
+
+	public static final String							FACTORY_CONFIGURATION_PID	= "my.factory.configuration.pid";
+	public static final String							NONSTATIC_CONFIGURATION_PID	= "nonstatic.configuration.pid";
+
+	private static ServiceReference<ConfigurationAdmin>	sref						= null;
+	private static BundleContext						bundleContext				= null;
+	private static ConfigurationAdmin					ca							= null;
+
+	@BeforeAll
+	public static void beforeAll() {
+
+		bundleContext = FrameworkUtil.getBundle(ConfigAnnotationTest.class)
+			.getBundleContext();
+
+		sref = bundleContext.getServiceReference(ConfigurationAdmin.class);
+		if (sref == null) {
+			throw new IllegalStateException("The configuration admin service cannot be found.");
+		}
+
+		ca = bundleContext.getService(sref);
+		if (ca == null) {
+			throw new IllegalStateException("The configuration admin service cannot be found.");
+		}
+
+	}
+
+	@AfterAll
+	public static void afterAll() {
+		BundleContext bundleContext = FrameworkUtil.getBundle(ConfigAnnotationTest.class)
+			.getBundleContext();
+		if (bundleContext != null && sref != null) {
+			bundleContext.ungetService(sref);
+		}
+
+	}
+
+	// START TESTS
+	@InjectConfiguration(NONSTATIC_CONFIGURATION_PID)
+	Configuration nonStaticConfiguration;
+
+	@Test
+	public void testFieldConfiguration() throws Exception {
+
+		Configuration cs = ConfigUtil.getConfigsByServicePid(ca, NONSTATIC_CONFIGURATION_PID);
+
+		assertThat(cs).isEqualTo(nonStaticConfiguration);
+
+	}
+
+	static final String PARAM_PID = "param.pid";
+
+	@Test
+	@WithConfiguration(pid = PARAM_PID, properties = {
+		@ConfigEntry(key = "bar", value = "foo")
+	})
+	public void testParameterConfiguration(
+		@InjectConfiguration(PARAM_PID) Configuration configuration)
+		throws Exception {
+
+		Configuration cs = ConfigUtil.getConfigsByServicePid(ca, PARAM_PID);
+		assertThat(cs).isEqualTo(configuration);
+
+		assertNull(configuration.getProperties()
+			.get("foo"));
+
+		String foo = (String) configuration.getProperties()
+			.get("bar");
+		assertEquals("foo", foo);
+	}
+
+	@Test
+	@WithConfiguration(pid = PARAM_PID, properties = {
+		@ConfigEntry(key = "foo", value = "bar")
+	})
+	public void testParameterConfiguration2(
+		@InjectConfiguration(PARAM_PID) Configuration configuration)
+		throws Exception {
+
+		Configuration cs = ConfigUtil.getConfigsByServicePid(ca, PARAM_PID);
+		assertThat(cs).isEqualTo(configuration);
```

Comment:
Configuration.equals just compares the configuration pids. Not the configuration properties. Is that what you mean here?

Reply:
done